conbench/entities/history.py

```diff
@@ -419,27 +379,23 @@ def _query_and_calculate_distribution_stats(
         .filter(BenchmarkResult.error.is_(None), Hardware.hash == hardware_checksum)
     )
 
-    # Filter to the correct case(s)/context(s)
-    if case_id and context_id:
-        # filter to the given case/context
+    # Filter to the correct history fingerprint(s)
+    if history_fingerprint:
+        # filter to the given history_fingerprint
         history = history.filter(
-            BenchmarkResult.case_id == case_id,
-            BenchmarkResult.context_id == context_id,
+            BenchmarkResult.history_fingerprint == history_fingerprint
```

Comment:
I was wondering why hardware wan't in here, then realized it's filtered in the query above. Can't we skip the hardware filtering there now, since this (or the version in the `else` clause below) will handle it already? 

I guess we still need to join to `hardware` to get `hardware_hash`, annoyingly; I wish that was just the ID so it was already on `benchmark_result`.

Reply:
Ah yeah, we could skip the filter above. That's nice.